scripts/generate-docs.ts

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+import * as fs from "fs-extra";
+import * as _ from "lodash";
+
+const program = require("commander");
+const pkg = require("../package.json");
+
+// TODO: Don't want to do lib
+const loadAllCommands = require('../lib/commands');
+
+program.version(pkg.version);
+
+var client: any = {
+    cli: program,
+    getCommand: function(name: any) {
+        for (var i = 0; i < client.cli.commands.length; i++) {
+            if (client.cli.commands[i]._name === name) {
+                return client.cli.commands[i];
+            }
+        }
+        return null;
+    },
+}; 
+
+loadAllCommands(client);
+
+function listCommands() {
+    return client.cli.commands.map((c: any) => {
+        return c.name();
+    });
+}
+
+function commandPagePage(name: string) {
+    const fileName = name.replace(/:/g, '-') + '.md';
+    const fullPath = `./docs/${fileName}`;
+
+    return fullPath;
+}
+
+function generateCommandPage(name: string, command: any) {
+    return `
+# ${name}
+
+${_.upperFirst(command._description)}
+
+## Usage
+\`\`\`
+firebase ${name} ${command.usage()}
+\`\`\`
+
+## Options
+\`\`\`
+${command.optionHelp()}
+\`\`\`
+`.trimLeft()
```

Comment:
I would suggest bringing in something like `handlebars` or another (light-weight) templating language to deal with this. That would allow you to break the template out of this file, and gain the power of the templating language for any slightly-advanced stuff you may want.